### PR TITLE
Adds ListActions back in

### PR DESCRIPTION
**Pull request by thostetler**
- Updates to use store
- Gets select all/none working properly
- Refactor Sort component, get it working properly

- Adds Why did you render utility for debugging performance issues

### CODE REVIEW

File: src/store/slices/docs.ts [hunk 1]
```diff
@@ -61,5 +82,30 @@ export const docsSlice: StoreSlice<IAppStateDocsSlice> = (set, get) => ({
...
+      }),
+      false,
+      'docs/selectAll',
+    ),
+
+  // remove items that exist in both lists, but keep the others in `selected` intact
+  clearAllSelected: () =>
+    set(
+      (state) => ({
+        docs: { ...state.docs, selected: difference(state.docs.selected, state.docs.current), isAllSelected: false },
```
Comment (shinyichen) [inline src/store/slices/docs.ts:103]: I don't really understand the purpose of the function here. Is it suppose to clear all so nothing is selected? So `selected` should be empty?

**Review by shinyichen — COMMENTED** [review]
Just an observation, maybe it belongs to a separate PR. While using class form, if I change the sort, the results page is still sorted by date, although the URL shows the right sort I selected. Maybe related, the handleOnSubmit in the cassic-form is not being used.

File: src/store/slices/docs.ts (hunk 1 above)
Comment (thostetler) [reply src/store/slices/docs.ts:103]: So I should have thought up a better name/way to do it.

But the problem is that we want an overall selection that persists as you page through results, but have the `Select All` functionality work page to page.  So it should clear after each pagination.  So we want to only clear the docs that exist in both lists, but leave the rest of `selected` intact.
So in a way, there shouldn't be a time when we're completely clearing the selected docs (unless we are doing a new search, for example)

File: src/components/ResultList/ListActions.tsx [hunk 2]
```diff
@@ -1,7 +1,93 @@
...
+  const selectAll = useStore((state) => state.selectAll);
+  const clearAllSelected = useStore((state) => state.clearAllSelected);
+  const isAllSelected = useStore((state) => state.docs.isAllSelected);
+
+  return (
+    <>
+      <Button variant="link" fontWeight="normal" disabled={isAllSelected} onClick={selectAll}>
+        Select All
+      </Button>
+      <Button variant="link" fontWeight="normal" disabled={!isAllSelected} onClick={clearAllSelected}>
```
[the remark this answers is not in the document]
Comment (shinyichen) [reply src/components/ResultList/ListActions.tsx:86]: Select none is only enabled when all is selected. But should be enabled if anything on the page is selected. 
(In the mean time, I will be thinking about the UI for clearing all selections)
Screen Shot 2022-03-22 at 11 04 14 AM

**Pull request merged**